docs/spec_draft.md

```diff
@@ -2313,6 +2314,67 @@ where `predicate = rank(pred) == 0 ? pred : pred[i0, ..., iR-1]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.sign
+
+### Semantics
+
+Returns the sign of the `operand` element-wise and produces a `result` tensor,
+using [stablehlo.compare](#stablehlocompare) as the comparator. For each element
+`x`, it follows the following pseudocode:
+```
+if element_type(x) is signed integer:
+  if stablehlo.compare(lhs=x, rhs=0, comparison_direction=LT, compare_type=SIGNED):
+    return -1
+  if stablehlo.compare(lhs=x, rhs=0, comparison_direction=EQ, compare_type=SIGNED):
+    return 0
+  if stablehlo.compare(lhs=x, rhs=0, comparison_direction=GT, compare_type=SIGNED):
+    return 1
+else if element_type(x) is floating-point:
+  if x is NaN:
+    return NaN
+  else:
+    if stablehlo.compare(lhs=x, rhs=0.0, comparison_direction=LT, compare_type=FLOAT):
+      return -1.0
+    if stablehlo.compare(lhs=x, rhs=-0.0, comparison_direction=EQ, compare_type=FLOAT):
+      return -0.0
+    if stablehlo.compare(lhs=x, rhs=0.0, comparison_direction=EQ, compare_type=FLOAT):
+      return 0.0
+    if stablehlo.compare(lhs=x, rhs=0.0, comparison_direction=GT, compare_type=FLOAT):
+      return 1.0
+else if element_type(x) is complex:
+  if x.real is NaN or x.imag is NaN:
+    return NaN
+  else:
+    return x / stablehlo.abs(x)
+```
```

Comment:
Proposed rewording:

```
Returns the sign of the `operand` element-wise and produces a `result` tensor.
More formally, for each element `x`, the semantics can be expressed using 
Python-like syntax as follows:
```

```Python
def sign(x):
  if is_integer(x):
    if compare(x, 0, LT, SIGNED): return -1
    if compare(x, 0, EQ, SIGNED): return 0
    if compare(x, 0, GT, SIGNED): return 1
  elif is_float(x):
    if x is NaN:
      return NaN
    else:
      if compare(x, 0.0, LT, FLOAT): return -1.0
      if compare(x, -0.0, EQ, FLOAT): return -0.0
      if compare(x, +0.0, EQ, FLOAT): return +0.0
      if compare(x, 0.0, GT, FLOAT): return 1.0
  elif is_complex(x):
    if x.real is NaN or x.imag is NaN:
      return NaN
    else:
      return divide(x, abs(x))
```

Feedback:
  1. "using stablehlo.compare as the comparator" sounds like a detail that best belongs in the formalism.
  2. Let's use Python syntax - I made a few changes to the code to accommodate that: a) added a "def", b) changed "if element_type(x) is foo" to just "is_foo(x)", 3) changed "else if" to "elif".
  3. I've also compressed the code by dropping "stablehlo.", removing kwarg syntax (since constant names unambiguously point to which parameter is which) and moving most returns to the same line as ifs to simplify eyeballing.